inspirehep/modules/records/receivers.py

```diff
@@ -391,36 +394,8 @@ def check_if_record_is_going_to_be_deleted(sender, *args, **kwargs):
             soft_delete_pidstore_for_record(record.id)
 
 
-def populate_experiment_suggest(sender, json, *args, **kwargs):
-    """Populates experiment_suggest field of experiment records."""
-
-    # FIXME: Use a dedicated method when #1355 will be resolved.
-    if 'experiments.json' in json.get('$schema'):
-        experiment_names = get_value(json, 'experiment_names.title')
-        title_variants = force_force_list(
-            get_value(json, 'title_variants.title'))
-
-        json.update({
-            'experiment_suggest': {
-                'input': experiment_names + title_variants,
-                'output': experiment_names[0],
-                'payload': {'$ref': get_value(json, 'self.$ref')},
-            },
-        })
-
-
-def populate_abstract_source_suggest(sender, json, *args, **kwargs):
-    """Populate abstract_source_suggest field of HEP records."""
-
-    # FIXME: Use a dedicated method when #1355 will be resolved.
+def populate_suggesters(sender, json, *args, **kwargs):
     if 'hep.json' in json.get('$schema'):
-        abstracts = json.get('abstracts', [])
-        for abstract in abstracts:
-            source = abstract.get('source')
-            if source:
-                abstract.update({
-                    'abstract_source_suggest': {
-                        'input': source,
-                        'output': source,
-                    },
-                })
+        populate_abstract_suggester(json)
+    elif 'experiments.json' in json.get('$schema'):
+        populate_experiment_name_suggester(json)
```

Comment:
Anyway this PR was WIP to start the discussion and see if we find a better way of organizing that code, but can be closed if we don't see any improvement possible atm.